03-react-week-2/advanced-forms.md

```diff
@@ -49,13 +54,107 @@ const onInputChange = (event) => {
+* id: ebeeac6e-9dfa-4730-b3ad-4f6d3b2d382e
+* title: event.target.name
+* points: 1
+* topics: react, react-forms
+
+##### !question
+
+What is stored in `event.target.name`?
+##### !end-question
+
+##### !placeholder
+
+What is stored in event.target.name?
+
+##### !end-placeholder
+
+##### !answer
+
+/.+/
+
+##### !end-answer
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, hidden, students click to view) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !explanation
+
+When an input field is defined in the JSX with `<input name="example_name" value={value} onChange={eventHandler} />` and the callback function is called, an object `event` which describes the circumstates of the event is passed into the function as an argument.
+
+`event.target` is the DOM object which triggered the event.  `event.target.name` returns the `name` attribute of the `input` element, "example_name" in this case.
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: short-answer
+* id: 0330ed08-109a-4d18-832d-4bf3aea32af5
+* title: Why use the subscript notation?
+* points: 1
+* topics: react, react-forms
+
+##### !question
+
+Why do we need to use subscript notation (square brackets) to update the state? In the past we used dot notation, which was more concise. What has changed?
+
+##### !end-question
+
+##### !placeholder
+
+Why use []?
+
+##### !end-placeholder
+
+##### !answer
+
+/.+/
+
+##### !end-answer
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, hidden, students click to view) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !explanation
+
+Because if we use the `.` operator, we cannot use a variable name for the  state attribute to change.  Instead we can use the `[]` and put a variable in the middle.
+
+For example: 
+
+```javascript
+const newFormFields = {
+    ...formFields,
+  }
+
+newFormFields[event.target.name] = event.target.value;
+setFormFields(newFormFields);
+```
+
+This sets `newformFields` to the old value of the form state and then updates the one field that changed.
+
+##### !end-explanation
 
-**Question:** Why do we need to use subscript notation (square brackets) to update the state? In the past we used dot notation, which was more concise. What has changed?
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
 
 ### Wait what does `...formFields` do?
 
-The code below is a a bit of JavaScript syntax called the rest operator which allow us to duplicate the `formFields` object.  
+The code below is a a bit of JavaScript syntax called the **rest operator** which allow us to duplicate the `formFields` object.  
```

Comment:
This should be a **spread operator** right? Since a rest operator is only in function parameters?